src/coreclr/jit/fgopt.cpp

```diff
@@ -3306,8 +3306,6 @@ bool Compiler::fgReorderBlocks(bool useProfile)
 {
     noway_assert(opts.compDbgCode == false);
 
-    assert(UsesFunclets() == fgFuncletsCreated);
```

Comment:
I presume by the time `useProfile` is `true` this assert should be always true? not sure it makes sense to keep, though